objects/isn_unifiedgrowingtray.lua

```diff
@@ -0,0 +1,524 @@
+		if secondsThisUpdate < 0 then secondsThisUpdate = dt end
+	end
+	storage.lastWorldTime = world.time()
+	
+	--Don't allow infinite fast forward time.
+	if secondsThisUpdate > maxFastForwardSeconds then secondsThisUpdate = maxFastForwardSeconds end
+	
+	--Use this to fast forward in increments.  This is intentionally lossy since accuracy costs server load.
+	if secondsThisUpdate > fastForwardSeconds then
+		storage.lastWorldTime = storage.lastWorldTime - (secondsThisUpdate - fastForwardSeconds)
+		secondsThisUpdate = fastForwardSeconds
+	end
+	
+	--useful for debugging...
+	--sb.logInfo("[%s], growth %s/%s seconds %s", storage.currentseed.name, storage.growth, storage.growthCap, secondsThisUpdate)
+	
+	local growthmod = secondsThisUpdate
+	if powerReq > 0 then
+	--Adjust our fast forward time by available power...
+		local availPower = power.getTotalEnergy()
+		if powerReq * secondsThisUpdate > availPower and availPower >= minFastForwardSeconds * powerReq then
+			local pwrIncrement = availPower / powerReq
+			storage.lastWorldTime = storage.lastWorldTime - (secondsThisUpdate - pwrIncrement)
+			secondsThisUpdate = pwrIncrement
+			growthmod = secondsThisUpdate
+		end
+		if power.consume(powerReq * secondsThisUpdate) then
+			animator.setAnimationState("powlight", "on")
+		else
+			growthmod = growthmod * cUnpoweredGrowthRate()
+			animator.setAnimationState("powlight", "off")
+		end
+	end
+	
+	--Compute which graphic should be displayed and update.
+	local growthperc = isn_getXPercentageOfY(storage.growth, storage.growthCap)
+	if growthperc >= 75 then
+		animator.setAnimationState("growth", "3")
+	elseif growthperc >= 50 then
+		animator.setAnimationState("growth", "2")
+	elseif growthperc >= 25 then
+		animator.setAnimationState("growth", "1")
+	else
+		animator.setAnimationState("growth", "0")
+	end
+	
+	--Check if a previous fluid use failed...
+	if storage.hasFluid == false then storage.hasFluid = isn_doFluidConsume() end
+	if storage.hasFluid ~= true then return end	--Failed again?  No growth nor outputs.
+	
+	storage.activeConsumption = true
+	
+	--Compute growth (we aren't dry right now)
+	storage.growth = storage.growth + ((storage.growthRate + storage.growthFluid + storage.growthFert) * growthmod)
+	
+	--Check if we should try to use more fluid due to growth.
+	if storage.currentStage < storage.stages then
+		--multiple stages might have happened, make sure we use apt fluid before resuming flow.
+		while true do
+			if storage.currentStage >= storage.stages then break end
+			if storage.growth < storage.stage[storage.currentStage].val then break end
+			if storage.hasFluid == false then break end
+			storage.hasFluid = isn_doFluidConsume()
+			storage.currentStage = storage.currentStage + 1
+		end
+	end
+	
+	--If the above fluid use left us dry stop here (don't output items) and try again later.
+	--Yes this may generate 1 more (unnecessary) growth spurt but it won't use more fluid.
+	if storage.hasFluid ~= true then return end
```

Comment:
`if not storage.hasFluid then return end`
unless you have a specific reason to check for true specifically ???